packages/flutter/test/widgets/text_selection_test.dart

```diff
@@ -653,6 +653,47 @@ void main() {
 
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.android }));
 
+  testWidgets('test TextSelectionGestureDetectorBuilder shows spell check toolbar on single tap on iOS if word misspelled and text selection toolbar on additonal taps', (WidgetTester tester) async {
+    await pumpTextSelectionGestureDetectorBuilder(tester);
+    final FakeEditableTextState state = tester.state(find.byType(FakeEditableText));
+    final FakeRenderEditable renderEditable = tester.renderObject(find.byType(FakeEditable));
+    const TextSelection selection = TextSelection.collapsed(offset: 1);
+    state.updateEditingValue(const TextEditingValue(text: 'something misspelled', selection: selection));
+
+    // Mark word to be tapped as misspelled for testing.
+    state.markCurrentSelectionAsMisspelled = true;
+    await tester.pump();
+
+    // Test spell check suggestions toolbar is shown on first tap of misspelled word.
+    const Offset position = Offset(25.0, 200.0);
+    await tester.tapAt(position);
+    await tester.pumpAndSettle();
+
+    expect(state.showSpellCheckSuggestionsToolbarCalled, isTrue);
+
+    // Reset and test text selection toolbar is toggled for additional taps.
+    state.showSpellCheckSuggestionsToolbarCalled = false;
+    renderEditable.selection = selection;
+    await tester.pump(const Duration(seconds: 10));
+
+    // Test first tap.
+    await tester.tapAt(position);
+    await tester.pumpAndSettle();
+
+    expect(state.showSpellCheckSuggestionsToolbarCalled, isFalse);
+    expect(state.toggleToolbarCalled, isTrue);
+
+    // Reset and test second tap.
+    state.toggleToolbarCalled = false;
+    await tester.pump(const Duration(seconds: 10));
+
+    await tester.tapAt(position);
+    await tester.pumpAndSettle();
+
+    expect(state.showSpellCheckSuggestionsToolbarCalled, isFalse);
+    expect(state.toggleToolbarCalled, isTrue);
+    }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS }));
+
```

Comment:
Is it possible to do one test that does the full cycle of 1. tap misspelled word 2. tap suggestion in the toolbar 3. expect the suggestion to have replaced the misspelled word.

Reply:
This sort of behavior is tested in `editable_text_test.dart`, so I'll try adding a test for at least 2/3 there. And keep 1 tested here, like with the Material style toolbar.

Reply:
It looks fine to me now, thanks.